### ISSUE TITLE
Hovering over comments or "unrelated words" inside a block gives info about the function the block was passed to

### ISSUE THREAD

**Issue by qaisjp** [sorbet/sorbet#4184]
#### Input

[→ View on sorbet.run](https://sorbet.run/#%23%20typed%3A%20true%0A%0A%5B1%2C%202%5D.each%20do%20%7Cx%7C%0A%20%20puts%20x%0A%20%20%23%20Why%20does%20hovering%20over%20a%20comment%20give%20docs%20for%20%60Array%23each%60%3F%0A%0A%0A%20%20%23%20And%20hovering%20over%20the%20%60if%60%20or%20%60end%60%20below%20does%20it%20too!%0A%20%20if%20true%0A%20%20%20%20puts%20true%0A%20%20end%0Aend%0A)

```ruby
# typed: true

[1, 2].each do |x|
  puts x
  # Why does hovering over a comment give docs for `Array#each`?


  # And hovering over the `if` or `end` below does it too!
  if true
    puts true
  end
end
```

#### Observed output

![image](https://user-images.githubusercontent.com/923242/116582247-12340480-a90d-11eb-9e8a-3303ed49ce82.png)

#### Expected behavior

Hovering over a comment should not pop up with any information at all.

**Comment by jez** [comment]
This was fixed at some point, possibly in #5375 but I haven't confirmed that.

**Comment by qaisjp** [comment]
Thanks!
